src/lib.rs

```diff
@@ -73,12 +73,18 @@ fn load_certs_from_env() -> Option<Result<Vec<CertificateDer<'static>>, Error>>
 fn load_pem_certs(path: &Path) -> Result<Vec<CertificateDer<'static>>, Error> {
     let f = File::open(path)?;
     let mut f = BufReader::new(f);
-    rustls_pemfile::certs(&mut f)
+    let certs = rustls_pemfile::certs(&mut f)
         .collect::<Result<Vec<_>, _>>()
         .map_err(|err| {
             Error::new(
                 ErrorKind::InvalidData,
                 format!("could not load PEM file {path:?}: {err}"),
             )
-        })
+        })?;
+
+    let certs: Vec<CertificateDer<'static>> = certs
+        .into_iter()
+        .map(|c| c.into_owned())
+        .collect();
+    Ok(certs)
```

Comment:
All this reduces to

```rust
    rustls_pemfile::certs(&mut f)
        .map(|result| match result {
            Ok(der) => Ok(der),
            Err(err) => Err(Error::new(
                ErrorKind::InvalidData,
                format!("could not load PEM file {path:?}: {err}"),
            )),
        })
        .collect()
```

Reply:
Nice :-) Tidied up with your suggestion.